### ISSUE TITLE
调用 login 报错

### ISSUE THREAD

**Issue by iyuhang** [AgoraIO/Flutter-RTM#37]
最近每次调用 login 时都报错
但似乎功能又是正常的

await _client.login(null, id);

```
E/flutter (15834): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: NoSuchMethodError: The method 'call' was called on null.
E/flutter (15834): Receiver: null
E/flutter (15834): Tried calling: call(2, 1)
E/flutter (15834): #0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:53:5)
E/flutter (15834): #1      AgoraRtmClient._eventListener 
package:agora_rtm/src/agora_rtm_client.dart:113
E/flutter (15834): #2      _rootRunUnary  (dart:async/zone.dart:1134:38)
E/flutter (15834): #3      _CustomZone.runUnary  (dart:async/zone.dart:1031:19)
E/flutter (15834): #4      _CustomZone.runUnaryGuarded  (dart:async/zone.dart:933:7)
E/flutter (15834): #5      _BufferingStreamSubscription._sendData  (dart:async/stream_impl.dart:338:11)
E/flutter (15834): #6      _DelayedData.perform  (dart:async/stream_impl.dart:593:14)
E/flutter (15834): #7      _StreamImplEvents.handleNext  (dart:async/stream_impl.dart:709:11)
E/flutter (15834): #8      _PendingEvents.schedule.<anonymous closure>  (dart:async/stream_impl.dart:669:7)
E/flutter (15834): #9      _rootRun  (dart:async/zone.dart:1122:38)
E/flutter (15834): #10     _CustomZone.run  (dart:async/zone.dart:1023:19)
E/flutter (15834): #11     _CustomZone.runGuarded  (dart:async/zone.dart:925:7)
E/flutter (15834): #12     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:965:23)
E/flutter (15834): #13     _rootRun  (dart:async/zone.dart:1126:13)
E/flutter (15834): #14     _CustomZone.run  (dart:async/zone.dart:1023:19)
E/flutter (15834): #15     _CustomZone.runGuarded  (dart:async/zone.dart:925:7)
E/flutter (15834): #16     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:965:23)
E/flutter (15834): #17     _microtaskLoop  (dart:async/schedule_microtask.dart:43:21)
E/flutter (15834): #18     _startMicrotaskLoop  (dart:async/schedule_microtask.dart:52:5)
```

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.14.6 18G87, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.44.2)
[✓] Connected device (2 available)

**Comment by iyuhang** [comment]
![image](https://user-images.githubusercontent.com/14176635/81198476-1a0be200-8ff4-11ea-9b81-39057cd80407.png)

**Comment by iyuhang** [comment]
能看一下这是为什么吗
@plutoless @LichKing-2234 
```
E/flutter ( 2011):
E/flutter ( 2011): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: NoSuchMethodError: The method 'call' was called on null.
E/flutter ( 2011): Receiver: null
E/flutter ( 2011): Tried calling: call(28)
E/flutter ( 2011): #0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:53:5)
E/flutter ( 2011): #1      AgoraRtmChannel._eventListener 
package:agora_rtm/src/agora_rtm_channel.dart:78
E/flutter ( 2011): #2      _rootRunUnary  (dart:async/zone.dart:1134:38)
E/flutter ( 2011): #3      _CustomZone.runUnary  (dart:async/zone.dart:1031:19)
E/flutter ( 2011): #4      _CustomZone.runUnaryGuarded  (dart:async/zone.dart:933:7)
E/flutter ( 2011): #5      _BufferingStreamSubscription._sendData  (dart:async/stream_impl.dart:338:11)
E/flutter ( 2011): #6      _DelayedData.perform  (dart:async/stream_impl.dart:593:14)
E/flutter ( 2011): #7      _StreamImplEvents.handleNext  (dart:async/stream_impl.dart:709:11)
E/flutter ( 2011): #8      _PendingEvents.schedule.<anonymous closure>  (dart:async/stream_impl.dart:669:7)
E/flutter ( 2011): #9      _rootRun  (dart:async/zone.dart:1122:38)
E/flutter ( 2011): #10     _CustomZone.run  (dart:async/zone.dart:1023:19)
E/flutter ( 2011): #11     _CustomZone.runGuarded  (dart:async/zone.dart:925:7)
E/flutter ( 2011): #12     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:965:23)
E/flutter ( 2011): #13     _rootRun  (dart:async/zone.dart:1126:13)
E/flutter ( 2011): #14     _CustomZone.run  (dart:async/zone.dart:1023:19)
E/flutter ( 2011): #15     _CustomZone.runGuarded  (dart:async/zone.dart:925:7)
E/flutter ( 2011): #16     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:965:23)
E/flutter ( 2011): #17     _microtaskLoop  (dart:async/schedule_microtask.dart:43:21)
E/flutter ( 2011): #18     _startMicrotaskLoop  (dart:async/schedule_microtask.dart:52:5)
E/flutter ( 2011):
```

**Comment by iyuhang** [comment]
![image](https://user-images.githubusercontent.com/14176635/81257377-af938a00-9065-11ea-84d9-14af5ab642b1.png)

调用到 login 时报错：

```
E/flutter ( 9587): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: NoSuchMethodError: The method 'call' was called on null.
E/flutter ( 9587): Receiver: null
E/flutter ( 9587): Tried calling: call(2, 1)
E/flutter ( 9587): #0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:53:5)
E/flutter ( 9587): #1      AgoraRtmClient._eventListener 
package:agora_rtm/src/agora_rtm_client.dart:113
E/flutter ( 9587): #2      _rootRunUnary  (dart:async/zone.dart:1134:38)
E/flutter ( 9587): #3      _CustomZone.runUnary  (dart:async/zone.dart:1031:19)
E/flutter ( 9587): #4      _CustomZone.runUnaryGuarded  (dart:async/zone.dart:933:7)
E/flutter ( 9587): #5      _BufferingStreamSubscription._sendData  (dart:async/stream_impl.dart:338:11)
E/flutter ( 9587): #6      _DelayedData.perform  (dart:async/stream_impl.dart:593:14)
E/flutter ( 9587): #7      _StreamImplEvents.handleNext  (dart:async/stream_impl.dart:709:11)
E/flutter ( 9587): #8      _PendingEvents.schedule.<anonymous closure>  (dart:async/stream_impl.dart:669:7)
E/flutter ( 9587): #9      _rootRun  (dart:async/zone.dart:1122:38)
E/flutter ( 9587): #10     _CustomZone.run  (dart:async/zone.dart:1023:19)
E/flutter ( 9587): #11     _CustomZone.runGuarded  (dart:async/zone.dart:925:7)
E/flutter ( 9587): #12     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:965:23)
E/flutter ( 9587): #13     _rootRun  (dart:async/zone.dart:1126:13)
E/flutter ( 9587): #14     _CustomZone.run  (dart:async/zone.dart:1023:19)
E/flutter ( 9587): #15     _CustomZone.runGuarded  (dart:async/zone.dart:925:7)
E/flutter ( 9587): #16     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:965:23)
E/flutter ( 9587): #17     _microtaskLoop  (dart:async/schedule_microtask.dart:43:21)
E/flutter ( 9587): #18     _startMicrotaskLoop  (dart:async/schedule_microtask.dart:52:5)
E/flutter ( 9587):
```

**Comment by LichKing-2234** [comment]
能判断是哪个回调没执行么

**Comment by iyuhang** [comment]
@LichKing-2234 
我这里能获取到的错误信息好像只有上面的了

**Comment by iyuhang** [comment]
@LichKing-2234 
channel 现在当有事件来的时候也报错
```
E/flutter ( 9587): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: NoSuchMethodError: The method 'call' was called on null.
E/flutter ( 9587): Receiver: null
E/flutter ( 9587): Tried calling: call(21)
E/flutter ( 9587): #0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:53:5)
E/flutter ( 9587): #1      AgoraRtmChannel._eventListener 
package:agora_rtm/src/agora_rtm_channel.dart:78
E/flutter ( 9587): #2      _rootRunUnary  (dart:async/zone.dart:1134:38)
E/flutter ( 9587): #3      _CustomZone.runUnary  (dart:async/zone.dart:1031:19)
E/flutter ( 9587): #4      _CustomZone.runUnaryGuarded  (dart:async/zone.dart:933:7)
E/flutter ( 9587): #5      _BufferingStreamSubscription._sendData  (dart:async/stream_impl.dart:338:11)
E/flutter ( 9587): #6      _DelayedData.perform  (dart:async/stream_impl.dart:593:14)
E/flutter ( 9587): #7      _StreamImplEvents.handleNext  (dart:async/stream_impl.dart:709:11)
E/flutter ( 9587): #8      _PendingEvents.schedule.<anonymous closure>  (dart:async/stream_impl.dart:669:7)
E/flutter ( 9587): #9      _rootRun  (dart:async/zone.dart:1122:38)
E/flutter ( 9587): #10     _CustomZone.run  (dart:async/zone.dart:1023:19)
E/flutter ( 9587): #11     _CustomZone.runGuarded  (dart:async/zone.dart:925:7)
E/flutter ( 9587): #12     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:965:23)
E/flutter ( 9587): #13     _rootRun  (dart:async/zone.dart:1126:13)
E/flutter ( 9587): #14     _CustomZone.run  (dart:async/zone.dart:1023:19)
E/flutter ( 9587): #15     _CustomZone.runGuarded  (dart:async/zone.dart:925:7)
E/flutter ( 9587): #16     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:965:23)
E/flutter ( 9587): #17     _microtaskLoop  (dart:async/schedule_microtask.dart:43:21)
E/flutter ( 9587): #18     _startMicrotaskLoop  (dart:async/schedule_microtask.dart:52:5)
```
都是在 _eventListener 这
![image](https://user-images.githubusercontent.com/14176635/81261331-77dd1000-906e-11ea-8064-b91b6e5f1b33.png)

**Comment by iyuhang** [comment]
当有事件来时

> “Member left: 14953, channel: 1channel”


![image](https://user-images.githubusercontent.com/14176635/81261778-3d27a780-906f-11ea-8bd5-aa509b28afd5.png)

**Comment by LichKing-2234** [comment]
好的，我会尽快check

**Comment by iyuhang** [comment]
@LichKing-2234 老哥有结果了吗？

**Comment by LichKing-2234** [comment]
@iyuhang 看了下，是因为这些Function没有赋值，判空就可以了，不过应该不影响使用，后面会优化

**Comment by iyuhang** [comment]
@LichKing-2234 
能尽快修复一下吗，在大多数设备上没有问题，但在一些设备上会导致崩溃

**Comment by LichKing-2234** [comment]
@iyuhang crash log提供下 看看，可能不是这个地方的问题

**Comment by iyuhang** [comment]
我们也在排查，有不少用户反馈说启动崩溃，错误日志里只收集到了 rtm 这个错误，所以现在猜测有可能是这个导致的, 所以希望能把这种错误都处理一下
@LichKing-2234 

![image](https://user-images.githubusercontent.com/14176635/81368018-5426d200-9121-11ea-84d0-9070362dcf73.png)

![Screenshot_20200508-114512](https://user-images.githubusercontent.com/14176635/81368085-80dae980-9121-11ea-9700-31fb5c277585.jpg)

**Comment by LichKing-2234** [comment]
@iyuhang 你自己能在某些机型上复现crash么

**Comment by iyuhang** [comment]
@LichKing-2234 没有复现

**Comment by LichKing-2234** [comment]
@iyuhang 优化发版本可能要到下周了，你可以本地先判断下Function是否为null，然后测试发个版本看下会不会有问题

**Comment by plutoless** [comment]
点进去就闪退感觉和这个没啥关系，应该是别的错误？你们这边有做崩溃收集吗

**Comment by iyuhang** [comment]
@plutoless 
有做，用了 flutter_bugly 这个插件，但没有收集到崩溃

**Comment by pr-1** [comment]
I know how to fix this issue. You will have to follow the steps of events given on the official RTM docs. https://docs.agora.io/en/Real-time-Messaging/API%20Reference/RTM_java/index.html

![Screenshot 2020-05-24 at 2 23 15 PM](https://user-images.githubusercontent.com/26018750/82749926-3cf70e00-9dca-11ea-9390-a6aee3904314.png)



So you will have to create a function for each callback. 

_channel.onMemberCountUpdated = (int count) {} 
in your case
Whats Happening below is that when the user joins the channel. this callback is called internally by the RTM package and Since you don't have any methods to receive the callback it is giving null.
I face similar issue with onChannelStateChanged.  
I have used Google Translate to read some of your comments. I hope this sorts out the issue.

> 当有事件来时
> 
> > “Member left: 14953, channel: 1channel”
> 
> ![image](https://user-images.githubusercontent.com/14176635/81261778-3d27a780-906f-11ea-8bd5-aa509b28afd5.png)

**Comment by LichKing-2234** [comment]
fixed 0.9.14
